pkg/router/template/router.go

```diff
@@ -953,22 +966,63 @@ func generateDestCertKey(config *ServiceAliasConfig) string {
 }
 
 // getServiceUnits returns a map of service keys to their weights.
-// Weight suggests the % of traffic that a given service will receive
-// compared to other services pointed to by the route.
-func getServiceUnits(route *routeapi.Route) map[string]int32 {
+// The requests are loadbalanced among the services referenced by the route.
+// The weight (0-256, default 1) sets the relative proportions each
+// service gets (weight/sum_of_weights fraction of the requests).
+// When the weight is 0 no traffic goes to the service. If they are
+// all 0 the request is returned with 503 response.
+// For each service, the requests are distributed among the endpoints.
+// Each endpoint gets weight/numberOfEndpoints portion of the requests.
+// The above assumes roundRobin scheduling.
+func getServiceUnits(r *templateRouter, route *routeapi.Route) map[string]int32 {
 	serviceUnits := make(map[string]int32)
 	key := fmt.Sprintf("%s/%s", route.Namespace, route.Spec.To.Name)
+
+	// find the maximum weight
+	var maxWeight int32 = 1
+	if route.Spec.To.Weight != nil && *route.Spec.To.Weight > maxWeight {
+		maxWeight = *route.Spec.To.Weight
+	}
+	for _, svc := range route.Spec.AlternateBackends {
+		if svc.Weight != nil && *svc.Weight > maxWeight {
+			maxWeight = *svc.Weight
+		}
+	}
+	// scale the weights to near the maximum (256)
+	// This improves precision when scaling for the endpoints
+	var scaleWeight int32 = 256 / maxWeight
```

Comment:
@rajatchopra Thanks for taking a look. The idea of normalizing was to bring up the largest weight to a 3 digit number. In this 128 is OK. It just gives a little mote headroom.
For your 1.5% example, The weight, if not 0 should have a min of 1. Thanks for spotting this. I'll fix it.
With a min of 1, the number of pods can be reduced if the overall percent is still too high. The 20 pods could be reduced to maybe 3.